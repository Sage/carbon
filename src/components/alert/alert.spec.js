import React from 'react';
import { mount } from 'enzyme';
import Alert from '.';

describe('Alert', () => {
  let wrapper;
  const onCancel = jasmine.createSpy('cancel');

  beforeEach(() => {
    wrapper = mount(
      <Alert
        open
        onCancel={ onCancel }
        title='Alert title'
        subtitle='Alert Subtitle'
        data-element='bar'
        data-role='baz'
      />
    );
  });

  it('include correct component, element and role data tags', () => {
    wrapper = mount(
      <Alert
        open
        onCancel={ onCancel }
        title='Alert title'
        subtitle='Alert Subtitle'
        data-element='bar'
        data-role='baz'
      />
    );
    expect(wrapper.instance().props['data-element']).toEqual('bar');
    expect(wrapper.instance().props['data-role']).toEqual('baz');
  });

  describe('keyboard focus', () => {
    let mockEvent;

    beforeEach(() => {
      wrapper = mount(
        <Alert
          open title='Test'
          subtitle='Test' showCloseIcon={ false }
        />
      );

      mockEvent = {
        preventDefault() {}
      };

      jest.useFakeTimers();
    });

    it('does not remain on the dialog if close icon is shown', () => {
      wrapper.setProps({
        showCloseIcon: true
      });

      spyOn(mockEvent, 'preventDefault');

      wrapper.instance().onDialogBlur(mockEvent);
      expect(mockEvent.preventDefault).not.toHaveBeenCalled();
    });
  });
});
