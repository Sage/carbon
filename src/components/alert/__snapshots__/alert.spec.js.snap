// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert include correct component, element and role data tags 1`] = `
<Portal
  key="1"
>
  <div
    className="carbon-dialog"
    data-component="alert"
    data-element="bar"
    data-role="baz"
    data-state="open"
  >
    <CSSTransitionGroup
      component="div"
      transitionAppear={true}
      transitionAppearTimeout={500}
      transitionEnter={true}
      transitionEnterTimeout={500}
      transitionLeave={true}
      transitionLeaveTimeout={500}
      transitionName="modal-background"
    >
      <div
        className="carbon-modal__background"
      />
    </CSSTransitionGroup>
    <CSSTransitionGroup
      component="div"
      transitionAppear={true}
      transitionAppearTimeout={500}
      transitionEnter={true}
      transitionEnterTimeout={500}
      transitionLeave={true}
      transitionLeaveTimeout={500}
      transitionName="modal"
    >
      <styled.div
        aria-describedby="carbon-dialog-subtitle"
        aria-labelledby="carbon-dialog-title"
        data-component="alert"
        data-element="bar"
        data-role="baz"
        onBlur={[Function]}
        role="dialog"
        size="extra-small"
        style={
          Object {
            "minHeight": undefined,
          }
        }
        tabIndex={0}
        theme={
          Object {
            "colors": Object {
              "base": "#00A376",
              "border": "#668491",
              "brand": "#00DC00",
              "destructive": Object {
                "hover": "#9F2C3F",
              },
              "error": "#C7384F",
              "focus": "#FFB500",
              "focusedIcon": "#335B6D",
              "focusedLinkBackground": "#FFDA7F",
              "info": "#0073C2",
              "primary": "#26A826",
              "secondary": "#006300",
              "success": "#00B000",
              "tertiary": "#004400",
              "warning": "#E96400",
              "white": "#FFFFFF",
              "whiteMix": "#E5F4E5",
            },
            "disabled": Object {
              "background": "#E5EAEC",
              "border": "#CCD6DA",
              "button": "#E5EAEC",
              "buttonText": "rgba(0,0,0,.2)",
              "disabled": "rgba(0,0,0,0.55)",
              "input": "#F2F4F5",
              "text": "rgba(0,0,0,0.3)",
            },
            "icon": Object {
              "focus": "#335B6D",
            },
            "name": "base",
            "shadows": Object {
              "cards": "0 3px 3px 0 rgba(0,20,29,0.2), 0 2px 4px 0 rgba(0,20,29,0.15)",
              "depth1": "0 5px 5px 0 rgba(0,20,29,0.2), 0 10px 10px 0 rgba(0,20,29,0.1)",
              "depth2": "0 10px 20px 0 rgba(0,20,29,0.2), 0 20px 40px 0 rgba(0,20,29,0.1)",
              "depth3": "0 10px 30px 0 rgba(0,20,29,0.1), 0 30px 60px 0 rgba(0,20,29,0.1)",
              "depth4": "0 10px 40px 0 rgba(0,20,29,0.04), 0 50px 80px 0 rgba(0,20,29,0.1)",
            },
            "text": Object {
              "color": "rgba(0,0,0,0.9)",
              "placeholder": "rgba(0,0,0,0.3)",
              "size": "14px",
            },
          }
        }
      >
        <styled.div
          theme={
            Object {
              "colors": Object {
                "base": "#00A376",
                "border": "#668491",
                "brand": "#00DC00",
                "destructive": Object {
                  "hover": "#9F2C3F",
                },
                "error": "#C7384F",
                "focus": "#FFB500",
                "focusedIcon": "#335B6D",
                "focusedLinkBackground": "#FFDA7F",
                "info": "#0073C2",
                "primary": "#26A826",
                "secondary": "#006300",
                "success": "#00B000",
                "tertiary": "#004400",
                "warning": "#E96400",
                "white": "#FFFFFF",
                "whiteMix": "#E5F4E5",
              },
              "disabled": Object {
                "background": "#E5EAEC",
                "border": "#CCD6DA",
                "button": "#E5EAEC",
                "buttonText": "rgba(0,0,0,.2)",
                "disabled": "rgba(0,0,0,0.55)",
                "input": "#F2F4F5",
                "text": "rgba(0,0,0,0.3)",
              },
              "icon": Object {
                "focus": "#335B6D",
              },
              "name": "base",
              "shadows": Object {
                "cards": "0 3px 3px 0 rgba(0,20,29,0.2), 0 2px 4px 0 rgba(0,20,29,0.15)",
                "depth1": "0 5px 5px 0 rgba(0,20,29,0.2), 0 10px 10px 0 rgba(0,20,29,0.1)",
                "depth2": "0 10px 20px 0 rgba(0,20,29,0.2), 0 20px 40px 0 rgba(0,20,29,0.1)",
                "depth3": "0 10px 30px 0 rgba(0,20,29,0.1), 0 30px 60px 0 rgba(0,20,29,0.1)",
                "depth4": "0 10px 40px 0 rgba(0,20,29,0.04), 0 50px 80px 0 rgba(0,20,29,0.1)",
              },
              "text": Object {
                "color": "rgba(0,0,0,0.9)",
                "placeholder": "rgba(0,0,0,0.3)",
                "size": "14px",
              },
            }
          }
        >
          <Heading
            divider={true}
            separator={false}
            subheader="Alert Subtitle"
            subtitleId="carbon-dialog-subtitle"
            title="Alert title"
            titleId="carbon-dialog-title"
          />
        </styled.div>
        <styled.div>
          <styled.div />
        </styled.div>
        <Icon
          bgSize="small"
          className="carbon-dialog__close"
          data-element="close"
          onBlur={[Function]}
          onClick={[Function]}
          tabIndex="0"
          type="close"
        />
      </styled.div>
    </CSSTransitionGroup>
  </div>
</Portal>
`;
